### PR TITLE
allow multiple SitemapUrlProvider hosts

### PR DESCRIPTION
**Pull request by toooni**
International Projects often do have a multiple domain strategy (.com, .ch, .de ..). For this case i "replaced" the `host` config with `hosts`.
To avoid BC break, I kept the `host` configuration but had to make it optional. In my opinion it's alright because the command does throw an exception that no provider is registered. If somebody doesn't want to use the SitemapUrlProvider, it's the smoother solution anyway because `enable: false` is enough.

Maybe there's a chance the host config will be removed in a upcoming major version?

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -20,7 +20,10 @@ public function getConfigTreeBuilder()
                 ->arrayNode('sitemap_provider')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('host')->isRequired()->cannotBeEmpty()->end()
+                        ->arrayNode('hosts')
+                            ->prototype('scalar')->end()
+                        ->end()
+                        ->scalarNode('host')->end()
```
Comment (kbond) [inline DependencyInjection/Configuration.php]: Can you just add `->info('Deprecated')`

File: DependencyInjection/ZenstruckCacheExtension.php [hunk 2]
```diff
@@ -30,7 +30,12 @@ protected function loadInternal(array $mergedConfig, ContainerBuilder $container
         $container->getDefinition('zenstruck_cache.crawler')->replaceArgument(0, new Reference('zenstruck_cache.client'));
 
         if ($mergedConfig['sitemap_provider']['enabled']) {
-            $container->setParameter('zenstruck_cache.sitemap_provider.host', $mergedConfig['sitemap_provider']['host']);
+
+            if (isset($mergedConfig['sitemap_provider']['host'])) {
+                $mergedConfig['sitemap_provider']['hosts'] = array_merge($mergedConfig['sitemap_provider']['hosts'], array($mergedConfig['sitemap_provider']['host']));
+            }
+
+            $container->setParameter('zenstruck_cache.sitemap_provider.hosts', $mergedConfig['sitemap_provider']['hosts']);
```
Comment (kbond) [inline DependencyInjection/ZenstruckCacheExtension.php]: I think we should throw an exception is no hosts are set.

File: UrlProvider/SitemapUrlProvider.php [hunk 3]
```diff
@@ -18,13 +18,13 @@ class SitemapUrlProvider implements UrlProvider
      * @param string $host
```
Comment (kbond) [inline UrlProvider/SitemapUrlProvider.php]: `array`

File: UrlProvider/SitemapUrlProvider.php [hunk 4]
```diff
@@ -18,13 +18,13 @@ class SitemapUrlProvider implements UrlProvider
...
      * @param Client $client
      */
-    public function __construct($host, Client $client)
+    public function __construct($hosts, Client $client)
```
Comment (kbond) [inline UrlProvider/SitemapUrlProvider.php]: can you typehint `array`

File: Tests/DependencyInjection/ZenstruckCacheExtensionTest.php [hunk 5]
```diff
@@ -36,19 +36,20 @@ public function testWithHost()
         $this->assertTrue($this->container->has('zenstruck_cache.sitemap_provider'));
     }
 
-    /**
-     * @expectedException \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException
-     */
-    public function testMissingHost()
+    public function testWithHosts()
```
Comment (kbond) [inline Tests/DependencyInjection/ZenstruckCacheExtensionTest.php]: Can you add a test for just adding `host` to ensure BC? Also, can you keep the exception test if `hosts` and `host` isn't defined?

**Comment by kbond** [comment]
Looks good! Just a few small things and I will merge.

**Comment by kbond** [comment]
actually... technically v2.0.0 is still in alpha so... you can remove the `host` option.

**Comment by toooni** [comment]
thank you for the fast response and the nice bundle btw ;)

- i removed the `host` config now completely and added an entry in the changelog
- made sure that the new `hosts` config can not be empty
- wrote a test to test the empty `hosts` config

**Comment by kbond** [comment]
Awesome!

**Pull request merged**